### ISSUE TITLE
Stop sending ctrl-C/ctrl-Z to the job when attached

### ISSUE THREAD

**Issue by artemlops** [neuro-inc/platform-client-python#2061]
Problem: you almost never know if you are attached to a job or you're reading its logs. If reading logs, then you can disconnect yourself via pressing ctrl-C, however if you're attached, this command will be sent to the job's process and terminate it. I have already lost my progress a few times by accidentally killing my job while trying to detach from it.

Here I propose not to send ctrl-C / ctrl-Z to the job when attached, but instead detach terminal. Is there any use-case when it's useful to send interrupt/kill to the job's process?

**Comment by romasku** [comment]
I do not like the idea, sorry. 
All cases when you want to use `^C` locally also apply to a job. For example, you can run `neuro logs` inside the job, or run a `top` or just `grep -r` something too big. Attach is almost like an `ssh`, and it would be strange if ssh did not allow to send a signal to the process. I often use `neuro run -t ubuntu bash` and it will make it unusable.

On other hand, it can be good to make logs distinguishable from attach to avoid confusion. I see the following options:
- Drop support of streamed logs. It will help, but probably not a good idea to lose a feature.
- Automatically open logs with a pager (less). Then logs will be visually distinguishable, but will not scroll to the bottom automatically. 
- Add a footer to logs when accessed from TTY. Something like this:
```
job logs here 1
job logs here 2
job logs here 3
<new lines will be added here, the header will remain below>
--------------
You are viewing stream logs of job <job-id>
```


Thoughts?

**Comment by artemlops** [comment]
> Drop support of streamed logs

IDK this is the feature I always use to track job's health. On the other hand, I have no real use-case for sending ctrl-C to the job.

**Comment by artemlops** [comment]
Let's ask our internal team to vote here with :+1: or :-1:

**Comment by artemlops** [comment]
My main concern is that kubectl + docker support attach command, but they detach when `ctrl-C` was pressed. The same is done for SSH -- they don't kill `bash` process that the job is running. Thus, I expect behaviour to be unexpected for the majority of users.

**Comment by artemlops** [comment]
I totally agree that when you run a process **inside attach/exec** session, you should be able to interrupt it via pressing `ctrl-C`. However, I think this does not kill the root process of the job.

**Comment by artemlops** [comment]
The solution that will totally satisfy me if we implement a protection for the root process `init 1`.

**Comment by romasku** [comment]
I checked the code and found the following:
- For tty jobs, e.g. started with `-t` (on by default for tty usage, e.g not in scripts) we do not kill jobs for running `^C` - we just pass the corresponding byte ('\x04') to the jobs process stdin. So it's no UNIX signal sent. So for if a process does not read from terminal, it is ignored. Also, some processes can handle such byte differently:
```
neuro run -t ubuntu bash
...
root@job-28098a0e-02f4-4445-983a-a920a9455e88:/# ^C
root@job-28098a0e-02f4-4445-983a-a920a9455e88:/# 
```
- For non-tty jobs (`--no-tty`) that are started from tty we have a special prompt:
```
$ neuro run --no-tty ubuntu bash
√ Job ID: job-6156dfbe-201c-445d-b4f1-03c151ad3dfc
...
√ ===== Job is running, press Ctrl-C to detach/kill =====
  Interrupted. Please choose the action:
Ctrl-C or C -- Kill
Ctrl-D or D -- Detach
Enter or ESC -- Continue the attached mode
```
- For non-tty jobs (`--no-tty`) that are accessed not from a tty (for example, if you do `neuro run ... | less`) we **do kill** without any prompt. If it is your case, I agree and +1 to change behaviour to detach.

**Comment by romasku** [comment]
Resolution after discussion in slack:
We should replace automatic kill with detaching for non-tty jobs accessed not for tty: the closing `neuro run ... | tee` should not kill a job.

**Comment by romasku** [comment]
It's been some time since #2063 was merged. @artemlops do you still experience issues with unintended job cancellation?
